torch/testing/_internal/common_device_type.py

```diff
@@ -476,15 +476,24 @@ def instantiate_device_type_tests(generic_test_class, scope, except_for=None, on
     generic_members = set(generic_test_class.__dict__.keys()) - set(empty_class.__dict__.keys())
     generic_tests = [x for x in generic_members if x.startswith('test')]
 
+    # Derive defaults from environment variables if available, default is still none
+    # Usage:
+    # export PYTORCH_TESTING_DEVICE_ONLY_FOR=cuda,cpu
+    # export PYTORCH_TESTING_DEVICE_EXCEPT_FOR=xla
```

Comment:
good to explain all the current supported device_type, or how to find them (i myself find it hard to expect what is going to be run using EXCEPT_FOR=xla)

Reply:
For PyTorch proper there's only actually two device types that are available, `cuda,cpu`

I think however there's documentation on the top of this file that outlines all of the possible device types